src/zod-factory.spec.ts

```diff
@@ -0,0 +1,361 @@
+import { describe, expect, it } from 'vitest';
+import { z } from 'zod';
+import { createFactoryFromZod } from './zod-factory.js';
+
+describe('createFactoryFromZod', () => {
```

Comment:
please merge both test files into one test file - same name as the actual file with .spec.ts, e.g. zod.spec.ts.

Also make sure to cover ALL the examples in tests - i.e. whatever you give as example. we should have a test case validating this use case.